src/http/ngx_http_mruby_module.c

```diff
@@ -439,6 +462,12 @@ static char *ngx_http_mruby_merge_srv_conf(ngx_conf_t *cf, void *parent, void *c
     return NGX_CONF_ERROR;
 #endif
   }
+  if (conf->ssl_verify_client_code != NGX_CONF_UNSET_PTR || conf->ssl_verify_client_inline_code != NGX_CONF_UNSET_PTR) {
+    if (sscf->verify) {
+      conf->ssl_verify_client_prev_cb = SSL_CTX_get_verify_callback(sscf->ssl.ctx);
+      SSL_CTX_set_verify(sscf->ssl.ctx, SSL_VERIFY_PEER, (SSL_verify_cb)ngx_http_mruby_ssl_x509_verify_handler);
```

Comment:
In here I think we should follow what value set in `$ssl_verify_client` otherwise it will strictly check client certificate.
So, if `$ssl_verify_client` is optional, or opional_no_ca should be `SSL_VERIFY_PEER|SSL_VERIFY_FAIL_IF_NO_PEER_CERT` I think.